game-backend/src/helper/gameObjects.test.ts

```diff
@@ -0,0 +1,81 @@
+import StateBuilder from '../test-helper/StateBuilder';
+import { State } from '../../../darwin-types/State';
+import { GAME_OBJECT_TYPES } from '../../../darwin-types/game-objects/GameObject';
+import { getGameObjectsPerType } from './gameObjects';
+
+describe('getGameObjectsPerType', () => {
+  const FOOD_ID = 'FOODID1';
+  const UNIT_ID = 'unit1';
+  const multipleObjectsCount = 5;
+  let emptyState: State;
+  let oneObjectOfEachState: State;
+  let multipleObjectState: State;
+  let multipleObjectStateBuilder: StateBuilder;
+
+  beforeEach(() => {
+    oneObjectOfEachState = StateBuilder.buildState()
+      .addFood({ id: FOOD_ID, x: 5, y: 17 })
+      .addUnit({ id: UNIT_ID, x: 2, y: 10 })
+      .build();
+
+    multipleObjectStateBuilder = StateBuilder.buildState();
+    for (let i = 1; i <= multipleObjectsCount; i++) {
+      multipleObjectStateBuilder.addFood({ id: `Food_${i}`, x: i, y: i });
+      multipleObjectStateBuilder.addUnit({
+        id: `Unit_${i}`,
+        x: i + 1,
+        y: i + 1,
+      });
+    }
+    multipleObjectState = multipleObjectStateBuilder.build();
+
+    emptyState = StateBuilder.buildState().build();
+  });
+
+  it('filters state with zero food and units', () => {
```

Comment:
Write the smallest meaningful assertions.
In this case I'd propose to split this test into two: one for units and one for foods (same for the others)